app/models/visit.rb

```diff
@@ -40,4 +47,41 @@ def adult?(visitor)
   def same_visit?(other)
     other.visit_id == self.visit_id
   end
+
+  def log_any_missing_information
+    prefix = "Visit #{visit_id} is missing a"
+    log_missing_prisoner(prefix)
+    log_missing_prisoner_number(prefix)
+    log_missing_prison_name(prefix)
+    log_missing_prison(prefix)
+    log_missing_prison_slots(prefix)
+  end
+
+  private
+
+  def log_missing_prisoner(prefix)
+    return if prisoner.present?
+    Rails.logger.info("#{prefix} prisoner")
+  end
+
+  def log_missing_prisoner_number(prefix)
+    return if prisoner.blank? || prisoner.number.present?
+    Rails.logger.info("#{prefix} prisoner number")
+  end
+
+  def log_missing_prison_name(prefix)
+    return if prisoner.blank? || prison_name.present?
+    Rails.logger.info("#{prefix} prison name")
+  end
+
+  def log_missing_prison(prefix)
+    return if prisoner.blank? || prison.present?
+    Rails.logger.info("#{prefix} prison")
+  end
+
+  def log_missing_prison_slots(prefix)
+    return if prisoner.blank? || prison.blank?
+    return unless prison.slots.blank?
```

Comment:
This would only ever happen if the yaml file with the prison data in was merged with slot data removed for one or more prisons?